pycbc/inject.py

```diff
@@ -169,12 +172,44 @@ def apply(self, strain, detector_name, f_lower=None, distance_scale=1,
             signal_lal = signal.lal()
             add_injection(lalstrain, signal_lal, None)
             injection_parameters.append(inj)                            
-            
+            if strain.injcutter is not None and \
```

Comment:
Initially I was calling the injections from within the injection rejection code, but I think this is problematic. To do so requires either a) storing the full injection waveforms when they are generated the first time and then reusing them in the injection rejection code or b) regenerating the injections.

For a) the problem is memory. In O4 we might be having quite a number of long BNS injection waveforms (sampled at 16384Hz), which might last hundreds (or more!) seconds. Storing all of these could greatly increase the RAM footprint (and while one might argue that that could be done, you don't *have* to do that).

For b) the problem is CPU time. Some injection waveforms are slow to generate, and we cannot speed this up. A time-domain SEOBNRv3 waveform cannot be generated any other way. If I lower the sample rate, it fails. If I lower the cutoff frequency, it fails. So we would be doubling the CPU cost associated with injection generation. For SEOBNRv3 this is a significant fraction of the CPU cost for injection runs (and if we generate v3 waveforms from 15Hz it is a significant fraction of the full CPU cost). I really don't want to do this twice!

Therefore this ended up in the injection routine. I actually think it does belong there, having thought about this a bit more. Although maybe the make-truncated-injection code could be pulled off into a function called from within the main injection path in inject.py?